Game/MyProject/Game/GameStates/RaceState.cpp

```diff
@@ -30,10 +37,212 @@ void RaceState::onExit() {
+            score=0;
+            state_controller->requestChange(GOver);
+        }
+        else
+        {
+            frames++;
+            if(frames==10)
+            {
+                frames=0;
+             
+                auto val=raceMenu->getRowCol(score%10 +'0');
+        
+                std::string name="count"+std::to_string(noDigits(score%10));
+                const char* quadName=name.c_str();
+                
+                if(raceMenu->getScene()->getRoot()->findFirstNodeWithName("countRoot")->findFirstNodeWithName("count1")==nullptr && score==0)
+                {
+                    //rendering first quad
+                    std::vector<std::string> uniforms;
+                    uniforms.push_back("mTexture");
+                    uniforms.push_back("row");
+                    uniforms.push_back("column");
+                    
+                     std::shared_ptr<SceneNode> countScoreQuad=HelperManager::BuildTexturedQuad(quadName,HelperManager::CreateMeshInstance(meshManager.GetMesh("quad"),uniforms,"../../Game/BasicMenu/Shaders/AtlasSpriteShader"),"font.tga");
+                    
+                    countScoreQuad->setLocalTransform(IvVector3{17,0,9}, IvVector3{0,4.72,1}, IvVector3{2,2,2});
+                    raceMenu->getScene()->getRoot()->findFirstNodeWithName("countRoot")->addChild(countScoreQuad);
+                    
+                    raceMenu->getScene()->getRoot()->findFirstNodeWithName(quadName)->getRenderable()->setUniformValue(1, val.first);
+                    raceMenu->getScene()->getRoot()->findFirstNodeWithName(quadName)->getRenderable()->setUniformValue(2, val.second);
+                }
+                else if(raceMenu->getScene()->getRoot()->findFirstNodeWithName("countRoot")->findFirstNodeWithName("count1")==nullptr)
+                {
+                   //after pause state
+                   renderScore();
+                }
+                else
+                {
+                   raceMenu->getScene()->getRoot()->findFirstNodeWithName(quadName)->getRenderable()->setUniformValue(1, val.first);
+                   raceMenu->getScene()->getRoot()->findFirstNodeWithName(quadName)->getRenderable()->setUniformValue(2, val.second);
+                }
+                
+                if(noDigits(score)>noDigits(score-1))
+                    addNewDigit();
+                
+                checkNeedChangeDigit();
+    
+                score++;
+            }
+        }
+}   
+    
+
+bool RaceState::isPauseTriggered()
+{
+    return IvGame::mGame->mEventHandler->IsKeyPressed('x');
+}
+
+bool RaceState::isGameOverTriggered()
+{
+    return IvGame::mGame->mEventHandler->IsKeyPressed('q');
 }
 
-bool RaceState::isQuitTriggered() {
-    return IvGame::mGame->mEventHandler->IsKeyDown('q');
+unsigned int RaceState::noDigits(uint32_t number)
+{
+    if(number<10)
+        return 1;
+    else
+        return 1+noDigits(number/10);
+}
+
+void RaceState::addNewDigit()
```

Comment:
The entire logic for digits seems to be more specific for menus. Maybe they be moved there.